File: VB/XPCustomBindingExample/XPO/DatabaseSchemaHelper.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Xpo.Metadata

Namespace XPCustomBindingExample.XPO
	Public NotInheritable Class DatabaseSchemaHelper

		Private Sub New()
		End Sub

		Public Shared Function GetDefaultDatabaseSchema() As XPDictionary
			Dim dictionary As XPDictionary = New ReflectionDictionary()

			Dim ordersTable As XPClassInfo = dictionary.CreateClass("Order")
			ordersTable.CreateMember("ID", GetType(Integer), New KeyAttribute(True))
			ordersTable.CreateMember("ProductName", GetType(String))
			ordersTable.CreateMember("CustomerName", GetType(String))
			ordersTable.CreateMember("OrderDate", GetType(Date))
			ordersTable.CreateMember("Freight", GetType(Decimal))

			Return dictionary
		End Function
	End Class
End Namespace

File: CS/XPCustomBindingExample/XPO/DatabaseSchemaHelper.cs
using System;
using DevExpress.Xpo;
using DevExpress.Xpo.Metadata;

namespace XPCustomBindingExample.XPO {
    public static class DatabaseSchemaHelper {
        public static XPDictionary GetDefaultDatabaseSchema() {
            XPDictionary dictionary = new ReflectionDictionary();

            XPClassInfo ordersTable = dictionary.CreateClass("Order");
            ordersTable.CreateMember("ID", typeof(int), new KeyAttribute(true));
            ordersTable.CreateMember("ProductName", typeof(string));
            ordersTable.CreateMember("CustomerName", typeof(string));
            ordersTable.CreateMember("OrderDate", typeof(DateTime));
            ordersTable.CreateMember("Freight", typeof(decimal));

            return dictionary;
        }
    }
}